src/Config/Loader/XliffFileLoader.php

```diff
@@ -76,11 +76,40 @@ private function convertXlfToPhp($name, $language)
         $xml = $this->getDomDocumentFromFile($name);
 
         $return = "\n// ".str_replace(strtr(dirname($this->rootDir), '\\', '/').'/', '', strtr($name, '\\', '/'))."\n";
-        $units = $xml->getElementsByTagName('trans-unit');
+        $fileNodes = $xml->getElementsByTagName('file');
+        $language = strtolower($language);
+
+        /** @var \DOMElement[] $fileNodes */
+        foreach ($fileNodes as $fileNode) {
+            $tagName = 'target';
+
+            // Use the source tag if the source language matches
+            if (strtolower($fileNode->getAttribute('source-language')) === $language) {
+                $tagName = 'source';
+            }
```

Comment:
The `target-language` attribute is optional, so I don’t think this check is necessary. Also, we didn’t want to change the behavior of the current XLIFF loader in a patch release, but only fix a bug.